### ISSUE TITLE
globals in config doesn't work at all

### ISSUE THREAD

**Issue by spocke** [eslint/eslint#1054]
.eslintrc:

```
{
    "globals": {
        "x": true
    }
}
```

test.js:

```
(function() {
    "use strict";

    x();
})();
```

Command line:
eslint -c .eslintrc test.js

Result:
test.js
  4:1  error  'x' is not defined  no-undef

This should not produce an error since "x" is defined as a global.

However this works fine:
eslint --global x test.js
